### ISSUE TITLE
Emoji in headings leads to MD026 and MD051

### ISSUE THREAD

**Issue by lonix1** [DavidAnson/markdownlint#934]
Repro:
```md
# Title

## Contents

[Introduction](#introduction)
[Get started](#get-started-zap)                    <!-- MD051 -->

## Introduction

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
ut labore et dolore magna aliqua. 

## Get started :zap:                               <!-- MD026 -->

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
ut labore et dolore magna aliqua. 
```

That shows `MD051/link-fragments Link fragments should be valid` for the second TOC item. I pressed <kbd>ctrl .</kbd> which suggested `#get-started-⚡`, but that also reported the error. (Though I'd prefer the word instead.)

It also reports `MD026/no-trailing-punctuation: Trailing punctuation in heading [Punctuation: ':']` for the heading.

**Comment by DavidAnson** [comment]
You should not see either warning in the latest release of the library: [demo](https://dlaa.me/markdownlint/#%25m%23%20Title%0A%0A%23%23%20Contents%0A%0A%5BIntroduction%5D(%23introduction)%0A%5BGet%20started%5D(%23get-started-zap)%0A%0A%23%23%20Introduction%0A%0ALorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%2C%20sed%20do%20eiusmod%20tempor%20incididunt%0Aut%20labore%20et%20dolore%20magna%20aliqua.%0A%0A%23%23%20Get%20started%20%3Azap%3A%0A%0ALorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%2C%20sed%20do%20eiusmod%20tempor%20incididunt%0Aut%20labore%20et%20dolore%20magna%20aliqua.%0A). But note that the latest library version is not yet incorporated into CLI, CLI2, or the VS Code extension.
